tools/PHP_CLI_Helper.class.inc

```diff
@@ -0,0 +1,103 @@
+<?php
+
+/**
+ * Use bash to determine if certain unix tools are installed.  Which is used for
+ * generic system tools (e.g. wget), and dpkg if which fails.
+ *
+ * @param string $tool Name of tool
+ *
+ * @return bool representing if tool is installed
+ */
+function installed($tool) : bool
+{
+    // `which` returns empty if a tool is not installed.
+    // shell_exec captures this output.
+    if (shell_exec("which $tool")) {
+        return true;
+    }
+    // check installed pacakages with dpkg. Returns 0 if installed.
```

Comment:
Are you sure about that? dpkg(1) says:

```
EXIT STATUS         top

       0      The requested action was successfully performed.  Or a check
              or assertion command returned true.

       1      A check or assertion command returned false.

       2      Fatal or unrecoverable error due to invalid command-line
              usage, or interactions with the system, such as accesses to
              the database, memory allocations, etc.
```

(but doesn't really clarify what "a check or assertion command" is)

Reply:
Hmm well I mean the `-s` flag gives the status of a package and returns 1 when not found so I think it works in this case.